src/core/modals/consentModal.js

```diff
@@ -36,7 +36,6 @@ import { createPreferencesModal } from './preferencesModal';
  */
 const createFocusSpan = () => {
     const span = createNode('span');
-    span.tabIndex = -1;
```

Comment:
The removal of the `tabindex=-1` attribute renders the `span` element unfocusable (programmatically) and this causes a11y issues related to focus restoration.

Example issue:

0. By default when the page is loaded, the focus is on the `body` element
1. plugin runs with `autoShow: false`, (`body` element is still focused)
2. programmatically call `.showPreferences()` (preferencesModal recieves focus)
3. preferencesModal is hidden
4. focus is now restored to the last focused element (`body`)

this should technically work; focus *looks to be restored to the `body` element*, however, when you actually use `tab` to navigate the website, focus is set on the browers' tools/tabs, which is wrong. This weird focus behaviour was visible in chrome only.

The `tabindex=-1` on the span element fixes this bug: by focusing the span and then the last focused element, focus is properly restored to the body element (focus goes to the first focusable element inside the body, rather than outside).

https://github.com/orestbida/cookieconsent/blob/c5a64e239fe7bb710cfa6e0b8909bbab7e7135d8/src/core/api.js#L329-L332

Reply:
Thanks for the explanation, now I was able to reproduce the issue. I'll dig into it a bit more.

Reply:
The latest commit should do the trick while keeping the HTML valid for most of the time.

Reply:
Great!

Just did a minor refactoring.